src/Nethermind/Nethermind.Consensus/Tracing/Tracer.cs

```diff
@@ -33,14 +33,14 @@ private void Process(Block block, IBlockTracer blockTracer, IBlockchainProcessor
                We also want to make it read only so the state is not modified persistently in any way. */
 
             blockTracer.StartNewBlockTrace(block);
-
+            IWorldState? worldStateToUse = _worldStateProvider.GetGlobalWorldState(block.Header);
```

Comment:
how to deal with this
this is using worldStateProvider - so can you reset worldStateProvider directly and that resets all the worldState implementations